webapp/src/Controller/API/SubmissionController.php

```diff
@@ -278,6 +282,21 @@ public function addSubmissionAction(Request $request, ?string $id): Response
             throw new BadRequestHttpException(sprintf('User does not belong to a team'));
         }
 
+        if ($userId = $request->request->get('user_id')) {
+            // If the current user is an admin or API writer, allow it to specify the user
+            if ($this->isGranted('ROLE_API_WRITER')) {
+                // Load the user
+                /** @var User|null $user */
+                $user = $this->em->getRepository(User::class)->find($userId);
+
+                if (!$user || !$user->getTeam() || $user->getTeam()->getTeamid() !== $team->getTeamid()) {
```

Comment:
I know it's more code, but it would be more user friendly to expose which of these conditions failed

Reply:
I thought it more a security leak. Or do you think it does not matter? It's easy to add

Reply:
I don't think there's any info worth protecting in here.